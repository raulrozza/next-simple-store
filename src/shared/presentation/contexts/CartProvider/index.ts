export { default as useAddToCart } from './hooks/useAddToCart';
