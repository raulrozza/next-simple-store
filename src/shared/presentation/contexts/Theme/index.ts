export { default as useTheme } from './hooks/useTheme';
export { default as Theme } from './implementations/StyledComponentTheme';
