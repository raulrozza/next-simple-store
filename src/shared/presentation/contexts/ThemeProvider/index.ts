export { default as useTheme } from './hooks/useTheme';
export { default as ThemeProvider } from './implementations/StyledComponentTheme';
