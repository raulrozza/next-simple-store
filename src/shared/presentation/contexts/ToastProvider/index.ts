export { useToastProvider } from './context';
export { default as ToastProvider } from './implementations/ReactToastifyProvider';
