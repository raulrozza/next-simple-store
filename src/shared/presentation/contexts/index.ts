export * from './CartProvider';
export * from './ThemeProvider';
