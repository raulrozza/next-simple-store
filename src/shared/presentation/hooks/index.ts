import { trpc } from '@/shared/infra/http/trpc';

const { useQuery, useMutation } = trpc;

export { useQuery, useMutation };
