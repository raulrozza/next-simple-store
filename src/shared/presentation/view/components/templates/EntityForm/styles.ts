import styled, { css } from 'styled-components';

import { MAX_SCREEN_WIDTH } from '@/shared/presentation/constants/screen';
import { Card } from '@/shared/presentation/view/components/atoms';

export const Container = styled.section`
    display: flex;
    justify-content: center;
`;

export const Content = styled(Card)`
    ${({ theme }) => css`
        padding: ${theme.layout.spacing(4)};
        margin: ${theme.layout.spacing(1)};

        width: 100%;
        max-width: ${theme.layout.size(MAX_SCREEN_WIDTH)};
    `}
`;
