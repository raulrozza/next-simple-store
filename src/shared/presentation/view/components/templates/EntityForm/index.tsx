import { ComponentPropsWithoutRef, FC } from 'react';

import { Spacing } from '@/shared/presentation/view/components/atoms';
import { Form } from '@/shared/presentation/view/components/molecules';

import { Container, Content } from './styles';

interface EntityFormProps {
    title: string;
    fields: ComponentPropsWithoutRef<typeof Form>['fields'];
    button: ComponentPropsWithoutRef<typeof Form>['button'];
}

const EntityForm: FC<EntityFormProps> = ({ title, fields, button }) => (
    <Container>
        <Content as="main">
            <h1>{title}</h1>

            <Spacing size={4} />

            <Form fields={fields} button={button} />
        </Content>
    </Container>
);

export default EntityForm;
