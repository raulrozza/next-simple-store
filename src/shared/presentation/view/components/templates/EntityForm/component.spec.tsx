import { render, screen } from '@testing-library/react';
import { Formik } from 'formik';

import { ThemeProvider } from '@/shared/presentation/contexts';

import EntityForm from './index';

describe('EntityForm', () => {
    it('should render the component correctly, displaying the title and the form', () => {
        const view = render(
            <Formik initialValues={{}} onSubmit={jest.fn()}>
                <EntityForm
                    title="Form"
                    fields={[]}
                    button={{ text: 'Button' }}
                />
            </Formik>,
            {
                wrapper: ThemeProvider,
            },
        );

        expect(screen.getByText('Form')).toBeInTheDocument();
        expect(screen.getByText('Button')).toBeInTheDocument();
        expect(view).toMatchSnapshot();
    });
});
