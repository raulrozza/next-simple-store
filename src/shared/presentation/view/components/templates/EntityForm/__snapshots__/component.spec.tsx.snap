// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntityForm should render the component correctly, displaying the title and the form 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c4 {
  padding: 0.5rem 1rem;
  border-radius: 0.5rem;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.25rem;
  background-color: #754A56;
  color: #FFFFFF;
}

.c4:hover {
  background-color: #845864;
}

.c1 {
  background-color: #FFFFFF;
  border-radius: 1rem;
}

.c3 {
  margin-top: 1rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  padding: 1rem;
  margin: 0.25rem;
  width: 100%;
  max-width: 50rem;
}

<body>
    <div>
      <section
        class="c0"
      >
        <main
          class="c1 c2"
        >
          <h1>
            Form
          </h1>
          <div
            class="c3"
            size="4"
          />
          <form
            action="#"
          >
            <button
              class="c4"
              type="submit"
            >
              Button
            </button>
          </form>
        </main>
      </section>
    </div>
  </body>,
  "container": .c4 {
  padding: 0.5rem 1rem;
  border-radius: 0.5rem;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.25rem;
  background-color: #754A56;
  color: #FFFFFF;
}

.c4:hover {
  background-color: #845864;
}

.c1 {
  background-color: #FFFFFF;
  border-radius: 1rem;
}

.c3 {
  margin-top: 1rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  padding: 1rem;
  margin: 0.25rem;
  width: 100%;
  max-width: 50rem;
}

<div>
    <section
      class="c0"
    >
      <main
        class="c1 c2"
      >
        <h1>
          Form
        </h1>
        <div
          class="c3"
          size="4"
        />
        <form
          action="#"
        >
          <button
            class="c4"
            type="submit"
          >
            Button
          </button>
        </form>
      </main>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
